src/private_cuda.h

```diff
@@ -128,4 +140,14 @@ struct _gpukernel {
 #endif
 };
 
+struct _gpucomm {
+  cuda_context* ctx;  // Start after the context
+// #ifdef WITH_CUDA_NCCL
+  ncclComm_t c;
+// #endif  // WITH_CUDA_NCCL
```

Comment:
I mean having a char * field in cuda_context and setting the pointer with ncclGetErrorString().  This can also be extended for other things (like the blas errors).

I would like to avoid the macros inside cuda_context so that there won't be some cases where the struct will fit in the generic size and others where it doesn't.  I also don't feel that adding a status code for every cuda library in the context is a viable long-term approach.